### PR TITLE
Refactor StoragePerfTool

### PR DESCRIPTION
**Pull request by dangleptr**
1.  Support metrics such as QPS,  latency(p50/p90/99),  progress
2.  Use TokenBucket to control the speed,  now we could use just one thread to send about 100MB/s with 1KB payload.
3.  Optimize the information dumped. 

The output looks like
```
I0327 11:39:22.462410 49444 StoragePerfTool.cpp:109] Progress 0.949259%, qps=10639, latency(us) median = 813, p90 = 830, p99 = 832: Success [0]
I0327 11:39:23.960098 49444 StoragePerfTool.cpp:109] Progress 0.957261%, qps=10659.7, latency(us) median = 813, p90 = 830, p99 = 832: Success [0]
I0327 11:39:25.452921 49444 StoragePerfTool.cpp:109] Progress 0.965261%, qps=10693.8, latency(us) median = 812, p90 = 829, p99 = 831: Success [0]
I0327 11:39:26.943711 49444 StoragePerfTool.cpp:109] Progress 0.973215%, qps=10710.2, latency(us) median = 812, p90 = 829, p99 = 831: Success [0]
I0327 11:39:28.340073 49444 StoragePerfTool.cpp:109] Progress 0.980736%, qps=10707.4, latency(us) median = 813, p90 = 830, p99 = 832: Success [0]
I0327 11:39:29.742156 49444 StoragePerfTool.cpp:109] Progress 0.988294%, qps=10744.4, latency(us) median = 813, p90 = 832, p99 = 840: Success [0]
I0327 11:39:31.170789 49444 StoragePerfTool.cpp:109] Progress 0.996224%, qps=10778.9, latency(us) median = 812, p90 = 829, p99 = 846: Success [0]
I0327 11:39:32.575775 49444 StoragePerfTool.cpp:109] Progress 1.00389%, qps=10795, latency(us) median = 813, p90 = 857, p99 = 881: Success [0]
I0327 11:39:33.940294 49444 StoragePerfTool.cpp:109] Progress 1.01111%, qps=10783.5, latency(us) median = 813, p90 = 859, p99 = 880: Success [0]
I0327 11:39:35.349287 49444 StoragePerfTool.cpp:109] Progress 1.01881%, qps=10765.5, latency(us) median = 812, p90 = 829, p99 = 871: Success [0]
I0327 11:39:36.777081 49444 StoragePerfTool.cpp:109] Progress 1.02672%, qps=10787.1, latency(us) median = 821, p90 = 886, p99 = 896: Success [0]
I0327 11:39:38.182941 49444 StoragePerfTool.cpp:109] Progress 1.03443%, qps=10802, latency(us) median = 871, p90 = 912, p99 = 916: Success [0]
I0327 11:39:39.591694 49444 StoragePerfTool.cpp:109] Progress 1.04217%, qps=10831.6, latency(us) median = 871, p90 = 911, p99 = 915: Success [0]
I0327 11:39:40.972841 49444 StoragePerfTool.cpp:109] Progress 1.04953%, qps=10831.3, latency(us) median = 881, p90 = 910, p99 = 912: Success [0]
I0327 11:39:42.360219 49444 StoragePerfTool.cpp:109] Progress 1.0568%, qps=10811, latency(us) median = 878, p90 = 900, p99 = 902: Success [0]
I0327 11:39:43.789146 49444 StoragePerfTool.cpp:109] Progress 1.06459%, qps=10814.8, latency(us) median = 861, p90 = 881, p99 = 884: Success [0]
I0327 11:39:45.176132 49444 StoragePerfTool.cpp:109] Progress 1.07187%, qps=10820.8, latency(us) median = 851, p90 = 870, p99 = 873: Success [0]
I0327 11:39:46.558975 49444 StoragePerfTool.cpp:109] Progress 1.07911%, qps=10798.8, latency(us) median = 842, p90 = 861, p99 = 864: Success [0]
I0327 11:39:47.898666 49444 StoragePerfTool.cpp:109] Progress 1.0862%, qps=10787.1, latency(us) median = 834, p90 = 851, p99 = 854: Success [0]
I0327 11:39:49.276067 49444 StoragePerfTool.cpp:109] Progress 1.09345%, qps=10774, latency(us) median = 830, p90 = 846, p99 = 850: Success [0]
I0327 11:39:50.692018 49444 StoragePerfTool.cpp:109] Progress 1.10121%, qps=10741.8, latency(us) median = 829, p90 = 844, p99 = 848: Success [0]
I0327 11:39:52.086740 49444 StoragePerfTool.cpp:109] Progress 1.10862%, qps=10743.4, latency(us) median = 827, p90 = 843, p99 = 846: Success [0]
I0327 11:39:53.428001 49444 StoragePerfTool.cpp:109] Progress 1.11574%, qps=10734.5, latency(us) median = 822, p90 = 838, p99 = 841: Success [0]
I0327 11:39:54.790722 49444 StoragePerfTool.cpp:109] Progress 1.12291%, qps=10703.8, latency(us) median = 820, p90 = 833, p99 = 836: Success [0]
I0327 11:39:56.165609 49444 StoragePerfTool.cpp:109] Progress 1.13011%, qps=10691, latency(us) median = 820, p90 = 833, p99 = 836: Success [0]
I0327 11:39:57.511126 49444 StoragePerfTool.cpp:109] Progress 1.13723%, qps=10638.3, latency(us) median = 817, p90 = 831, p99 = 833: Success [0]
I0327 11:39:58.861101 49444 StoragePerfTool.cpp:109] Progress 1.14432%, qps=10600.6, latency(us) median = 813, p90 = 828, p99 = 830: Success [0]
I0327 11:40:00.230501 49444 StoragePerfTool.cpp:109] Progress 1.15147%, qps=10593.5, latency(us) median = 813, p90 = 828, p99 = 831: Success [0]
I0327 11:40:01.611985 49444 StoragePerfTool.cpp:109] Progress 1.15867%, qps=10579.1, latency(us) median = 813, p90 = 829, p99 = 831: Success [0]
I0327 11:40:02.973815 49444 StoragePerfTool.cpp:109] Progress 1.16583%, qps=10559.7, latency(us) median = 813, p90 = 829, p99 = 831: Success [0]
```

### CODE REVIEW

File: src/tools/storage-perf/StoragePerfTool.cpp [hunk 1]
```diff
@@ -196,91 +179,121 @@ class Perf {
...
         return edges;
     }
 
     void getNeighborsTask() {
         auto* evb = threadPool_->getEventBase();
-        std::vector<EdgeType> e(edgeType_);
-        auto f =
-          client_->getNeighbors(spaceId_, randomVertices(), std::move(e), "", randomCols())
+        auto tokens = tokenBucket_.consumeOrDrain(FLAGS_concurrency, FLAGS_qps, FLAGS_concurrency);
+        for (auto i = 0; i < tokens; i++) {
```
Comment (panda-sheep) [inline src/tools/storage-perf/StoragePerfTool.cpp]: I have a question,  when does the value of this `tokens ` not equal `FLAGS_concurrency`?

Comment (dangleptr) [reply src/tools/storage-perf/StoragePerfTool.cpp]: It depends the token rate.

Comment (critical27) [reply src/tools/storage-perf/StoragePerfTool.cpp:200]: The third parameter should be the qps?
> A token bucket stores up to a fixed number of tokens (the burst size)

Comment (critical27) [reply src/tools/storage-perf/StoragePerfTool.cpp:200]: Never mind

**Pull request merged**